packages/tabs/src/TabTitle.tsx

```diff
@@ -141,19 +165,32 @@ const TabTitle: ExtendableBox<BaseTabTitleProps, 'button'> = ({
   children,
   className,
   darkMode,
-  isAnyTabFocused,
+  parentRef,
   ...rest
 }: BaseTabTitleProps) => {
   const { usingKeyboard: showFocus } = useUsingKeyboardContext();
   const [showEllipsis, setShowEllipsis] = useState(false);
   const titleRef = useRef<HTMLAnchorElement | HTMLButtonElement>(null);
+  const activeEl = useDocumentActiveElement();
+
   const mode = darkMode ? Mode.Dark : Mode.Light;
 
+  // Checks to see if the current activeElement is a part of the same tab set
```

Comment:
Colocating focus logic. Right now, we are checking for tabsList.indexof(activeEl) inside of the Tabs component, and that useEffect was dependent on activeEl (so would rerun every time activeEl changed) and separately we were setting focus inside of this useEffect (so this useEffect would trigger the other, causing a bad side effect).

Reply:
Remind me why we need to manage focus of the tabs in the first place? I feel like we may be able to solve the problem by simplifying the implementation rather than adding complexity here.

Reply:
we need to manage focus manually because otherwise keyboard navigation won't focus a tab item.

Reply:
In the future, I think we should explore this pattern a bit more, but sounds good for now.